metadata/metadata.go

```diff
@@ -145,8 +153,12 @@ func NewOutgoingContext(ctx context.Context, md MD) context.Context {
 }
 
 // AppendToOutgoingContext returns a new context with the provided kv merged
-// with any existing metadata in the context. Please refer to the
-// documentation of Pairs for a description of kv.
+// with any existing metadata in the context. Please refer to the documentation
+// of Pairs for a description of kv.
+//
+// Unlike Pairs, the keys are not turned into lowercase. Users of
+// FromOutgoingContext need to handle them accordingly. Read
+// FromOutgoingContext's doc for more details.
```

Comment:
This isn't actually relevant / correct.  These keys will be lowercased in FromOutgoingContext.

Reply:
Removed